joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/util/ProgramHandlingUtil.scala

```diff
@@ -1,158 +1,136 @@
+            case Success(dir) => List(dir)
+            case Failure(e) =>
+              logger.warn(s"Failed to extract archive", e)
+              List.empty
+          })
+        case _ =>
+          Right(List.empty)
       }
-      destPath match {
-        case Some(destPath) =>
-          val dstFile = new File(destPath)
-          dstFile.mkdirs()
-          Files.copy(Paths.get(f), dstFile.toPath, StandardCopyOption.REPLACE_EXISTING)
-          Some(dstFile.getAbsolutePath)
-        case None => None
-      }
-    }
+    )
   }
 
-  /** Unzips a ZIP file into a sequence of files. All files unpacked are deleted at the end of CPG construction.
-    *
-    * @param zf
-    *   The ZIP file to extract.
-    * @param sourceCodePath
-    *   The project root path to unpack to.
-    */
-  def unzipArchive(zf: ZipFile, sourceCodePath: String): List[String] = {
-    val zipTempDir = Files.createTempDirectory("plume-unzip-")
-    try {
-      Using.resource(zf) { (zip: ZipFile) =>
-        // Copy zipped files across
-        return moveClassFiles(
-          zip
-            .entries()
-            .asScala
-            .filter(f => !f.isDirectory && f.getName.endsWith(".class"))
-            .flatMap(entry => {
-              val sourceCodePathFile = new File(sourceCodePath)
-              // Handle the case if the input source code path is an archive itself
-              val destFile = if (sourceCodePathFile.isDirectory) {
-                new File(zipTempDir.toAbsolutePath.toString + File.separator + entry.getName)
-              } else {
-                new File(zipTempDir.toAbsolutePath.toString + File.separator + entry.getName)
-              }
-              // dirName accounts for nested directories as a result of JAR package structure
-              val dirName = destFile.getAbsolutePath
-                .substring(0, destFile.getAbsolutePath.lastIndexOf(File.separator))
-              // Create directory path
-              new File(dirName).mkdirs()
-              try {
-                if (destFile.exists()) destFile.delete()
-                Using.resource(zip.getInputStream(entry)) { input =>
-                  Files.copy(input, destFile.toPath)
-                }
-                destFile.deleteOnExit()
-                Option(destFile.getAbsolutePath)
-              } catch {
-                case e: Exception =>
-                  logger
-                    .warn(
-                      s"Encountered an error while extracting entry ${entry.getName} from archive ${zip.getName}.",
-                      e
-                    )
-                  Option.empty
-              }
-            })
-            .toList
-        )
+  object ClassFile {
+    def getPackagePathFromByteCode(fis: FileInputStream): Option[String] = {
+      val cr                   = new ClassReader(fis)
+      var path: Option[String] = None
+      sealed class ClassNameVisitor extends ClassVisitor(Opcodes.ASM9) {
```

Comment:
I think my design for the initial `ClassNameVisitor` was pretty bad. How about moving `path` inside the class as a field:
```scala
sealed class ClassNameVisitor extends ClassVisitor(Opcodes.ASM9) {
    var path: Option[String] = None
    
    override def visit(...) { ... }
}

val rootVisitor = new ClassNameVisitor()
cr.accept(rootVisitor, SKIP_CODE)
rootVisitor.path
```
Just reduces the capturing of `path` in a separate closure in some way - also cleaner.